quickwit/quickwit-metastore/src/metastore/file_backed/mod.rs

```diff
@@ -715,26 +758,65 @@ impl MetastoreService for FileBackedMetastore {
             .with_limit(request.num_splits as usize);
         let list_splits_request =
             ListSplitsRequest::try_from_list_splits_query(&list_splits_query)?;
-        let splits = self.inner_list_splits(list_splits_request).await?;
+        let splits = self.list_splits_inner(list_splits_request).await?;
         ListSplitsResponse::try_from_splits(splits)
     }
 
     async fn index_metadata(
         &mut self,
         request: IndexMetadataRequest,
     ) -> MetastoreResult<IndexMetadataResponse> {
-        let index_id = request.get_index_id()?;
         let index_metadata = self
-            .read_any(&index_id, |index| Ok(index.metadata().clone()))
-            .await?;
-        if let Some(index_uid) = &request.index_uid {
-            if index_metadata.index_uid != *index_uid {
-                return Err(MetastoreError::NotFound(EntityKind::Index {
-                    index_id: index_id.to_string(),
-                }));
+            .index_metadata_inner(request.index_id, request.index_uid)
+            .await
+            .map_err(|(metastore_error, _index_id_opt, _index_uid_opt)| metastore_error)?;
+        let response = IndexMetadataResponse::try_from_index_metadata(&index_metadata)?;
+        Ok(response)
+    }
+
+    async fn indexes_metadata(
+        &mut self,
+        request: IndexesMetadataRequest,
+    ) -> MetastoreResult<IndexesMetadataResponse> {
+        let mut indexes_metadata: Vec<IndexMetadata> =
+            Vec::with_capacity(request.subrequests.len());
+        let mut failures: Vec<IndexMetadataFailure> = Vec::new();
+
+        let mut index_metadata_futures = FuturesUnordered::new();
```

Comment:
`FuturesUnordered` is useless in the case of the file backed metastore. The inner logic takes a mutex right away.

Reply:
`FuturesUnordered` is useless compared to? a for loop? Yes, each future acquires the outer lock but should work on a separate per-index inner lock. We want some level of concurrency there, don't we?

Reply:
You are right.